---
title: "FMT Model Analysis"
output: html_notebook
---

## Code flow and overview

To determine transfer efficiency between various microbial environments, we compared ASV abundance between samples, both within and across different FMT groups.  We performed Pearson's correlation coefficient test to describe the transfer efficiency between the two environments.
We initially used the non-parametric Spearman correlation test, but with non-transferred ASVs between samples presenting zero counts along each axis, the linearity of the transfer efficiency was best captured with the parametric Pearson correlation coefficient test. 

Transfer efficiency between each sample pair is visualized as a scatterplot; correlation coefficients across all possible sample pairs are summarized as boxplots.  

#### Data import and parsing

Running the "myInstalledPackages.R" script will install and import all relevant packages for the the following analyses.

The code below reads in the metadata and counts files with updated nomenclature; label changes were done in Excel. In addition to updating nomenclature, we formatted the sample names to make it R-friendly (i.e. change "-" into "." in sample names because R interprets "-" between integers as a mathematic operation).

There are a total of 110 samples. Counts tables (genus and ASV) are read in with a feature x sample format.

The metadata file is 110x8 (samples x feature). The ASV file is 4075x110 (feature x samples).

Before merging the counts table and metadata file (based on "SampleID" column), we first transposed the ASV counts table to maintain a consistent dataframe format (samples x feature).

```{r}

file_asv_updatedName = read.table("/Users/aduong/Desktop/Balfour2022/files/asv_biom-with-taxonomy_importToR_updatedNomenclature.txt", header = T, sep = "\t")
file_genusCountsTable_updatedName = read.table("/Users/aduong/Desktop/Balfour2022/files/feature-table_balfour_ver3_updatedNomenclature.txt", header = T, sep = "\t")
metamapping_updatedName = read.table("/Users/aduong/Desktop/Balfour2022/files/mappingMetadata_updatedNomenclature.txt", header = T, sep = "\t")
asv_taxonomy_map = read.table("files/asv_taxonomy_map.txt", sep = "\t", header = TRUE)

# need group labels, counts table WITH metadata, and a lognormalized counts table 
## group labels
labels_inputRecipientGroups = unique(metamapping_updatedName$FMTGroupFMTsource..Recipientbackground)

file_asv_updatedName_t = t(file_asv_updatedName)
file_asv_updatedName_t_samplesOnly = file_asv_updatedName_t[-(c(1,nrow(file_asv_updatedName_t))),]
colnames(file_asv_updatedName_t_samplesOnly) = file_asv_updatedName_t[1,]

#making this file as the metamapping with counts file :: this set is not used for lognormalizing, so adding in the SampleID column will not affect any downstream work
file_asv_updatedName_t_samplesOnly_toMerge = as.data.frame(file_asv_updatedName_t_samplesOnly)
file_asv_updatedName_t_samplesOnly_toMerge$SampleID = rownames(file_asv_updatedName_t_samplesOnly)

#merge counts and metadata files
myASVandMetadata_newNames = full_join(file_asv_updatedName_t_samplesOnly_toMerge, metamapping_updatedName, by="SampleID")

#column name that is specified in the getcorrelation functions
names(myASVandMetadata_newNames)[names(myASVandMetadata_newNames)=="FMTGroupFMTsource..Recipientbackground"] = "FMTGroupFMTsourcegtRecipientbackground"

##lognorm
file_asv_updatedName_t_samplesOnly_df = as.data.frame(file_asv_updatedName_t_samplesOnly)
file_asv_updatedName_t_samplesOnly_df[] <- lapply(file_asv_updatedName_t_samplesOnly_df, function(x) as.numeric(as.character(x)))

file_asv_updatedName_t_samplesOnly_lognorm = lognorm_function(file_asv_updatedName_t_samplesOnly_df)

## FILES
file_asv_updatedName_t_samplesOnly_lognorm #samples only, lognorm, samples x feature
labels_inputRecipientGroups #all labels
# serial passage groups
labels_NIMM_lineage = c("HM1input", "HM1->WT", "NIMM-g1input", "NIMM-g1->WT", "NIMM-g2input", "NIMM-g2->WT")
labels_IMM_lineage = c("HM1input", "HM1->KO", "IMM-g1input", "IMM-g1->KO","IMM-g2input", "IMM-g2->KO")

myASVandMetadata_newNames#counts table with metadata, not lognorm, samples x feature
```

The code below parses and log normalizes the genus counts table:

```{r}
## parse genus counts table
file_genusCountsTable_updatedName = read.table("/Users/aduong/Desktop/Balfour2022/files/feature-table_balfour_ver3_updatedNomenclature.txt", header = T, sep = "\t")
metamapping_updatedName = read.table("/Users/aduong/Desktop/Balfour2022/files/mappingMetadata_updatedNomenclature.txt", header = T, sep = "\t")

file_genusCountsTable_updatedName_t = t(file_genusCountsTable_updatedName)

genusCountsTable_samplebyfeatureFormat = file_genusCountsTable_updatedName_t[-1,]
colnames(genusCountsTable_samplebyfeatureFormat) = file_genusCountsTable_updatedName_t[1,]

            #######
#making this file as the metamapping with counts file :: this set is not used for lognormalizing, so adding in the SampleID column will not affect any downstream work
genusCounts_samplesOnly_toMerge = as.data.frame(genusCountsTable_samplebyfeatureFormat)
genusCounts_samplesOnly_toMerge$SampleID = rownames(genusCountsTable_samplebyfeatureFormat)

#merge counts and metadata files
genusCountsandMetadata = full_join(genusCounts_samplesOnly_toMerge, metamapping_updatedName, by="SampleID")

#column name that is specified in the getcorrelation functions
names(genusCountsandMetadata)[names(genusCountsandMetadata)=="FMTGroupFMTsource..Recipientbackground"] = "FMTGroupFMTsourcegtRecipientbackground"

# group labels for pcoa
metagroup = factor(genusCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground)

##lognorm
genusCounts_samplesOnly_df = as.data.frame(genusCountsTable_samplebyfeatureFormat)
genusCounts_samplesOnly_df[] <- lapply(genusCounts_samplesOnly_df, function(x) as.numeric(as.character(x)))

genusCounts_samplesOnly_df_lognorm = lognorm_function(genusCounts_samplesOnly_df)
```

Below, the code parses and log normalizes the phylum counts table:

```{r}
## parse phylum counts table
file_phylumCountsTable_updatedName = read.table("/Users/aduong/Desktop/Balfour2022/files/feature-table_balfour_phylum_updatedNomenclature.txt", header = T, sep = "\t")
metamapping_updatedName = read.table("/Users/aduong/Desktop/Balfour2022/files/mappingMetadata_updatedNomenclature.txt", header = T, sep = "\t")

file_phylumCountsTable_updatedName_t = t(file_phylumCountsTable_updatedName)

phylumCountsTable_samplebyfeatureFormat = file_phylumCountsTable_updatedName_t[-1,]
colnames(phylumCountsTable_samplebyfeatureFormat) = file_phylumCountsTable_updatedName_t[1,]

#######
#making this file as the metamapping with counts file :: this set is not used for lognormalizing, so adding in the SampleID column will not affect any downstream work
phylumCounts_samplesOnly_toMerge = as.data.frame(phylumCountsTable_samplebyfeatureFormat)
phylumCounts_samplesOnly_toMerge$SampleID = rownames(phylumCountsTable_samplebyfeatureFormat)

#merge counts and metadata files
phylumCountsandMetadata = full_join(phylumCounts_samplesOnly_toMerge, metamapping_updatedName, by="SampleID")

#column name that is specified in the getcorrelation functions
names(phylumCountsandMetadata)[names(phylumCountsandMetadata)=="FMTGroupFMTsource..Recipientbackground"] = "FMTGroupFMTsourcegtRecipientbackground"

# group labels for pcoa
metagroup = factor(phylumCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground)

##lognorm
phylumCounts_samplesOnly_df = as.data.frame(phylumCountsTable_samplebyfeatureFormat)
phylumCounts_samplesOnly_df[] <- lapply(phylumCounts_samplesOnly_df, function(x) as.numeric(as.character(x)))

phylumCounts_samplesOnly_df_lognorm = lognorm_function(phylumCounts_samplesOnly_df)
```

#### PCoA Plots

PCoA plots were generated with the capscale function from the vegan package using Bray-Curtis dissimilarity matrix.

```{r}
plotColors_ver2 = c("gray65", "royalblue1", "green", "orange", "purple",  
                    "black", "olivedrab", "red", "red4", 
                      "cyan","palevioletred1", "blue3", "yellow2", "magenta")  
                    
               
metagroup = factor(myASVandMetadata_newNames$FMTGroupFMTsourcegtRecipientbackground)

micegroups_all = c("HM1->KO", "HM1->WT", "IMM-g1->KO", "IMM-g2->KO", "NIMM-g1->WT","NIMM-g2->WT") 
micegroups_inflamed = c("HM1->KO", "IMM-g1->KO","IMM-g2->KO", "IMM-g1input", "IMM-g2input") 
micegroups_noninflamed = c("HM1->WT", "NIMM-g1->WT","NIMM-g2->WT", "NIMM-g1input", "NIMM-g2input") 
excludecrossoverGroup = unique(genusCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground)
excludecrossoverGroup = excludecrossoverGroup[-11]
micegroups_inflamed_withbothHMCohorts = c("HM1->KO", "HM2->KO","IMM-g1->KO","IMM-g2->KO", "IMM-g1input", "IMM-g2input") 
micegroups_serialTransfersPlusCrossover = c("HM1->KO", "HM1->WT", "IMM-g1->KO", "IMM-g2->KO", "NIMM-g1->WT","NIMM-g2->WT", "NIMM-g1->KO")

micegroups_all.samplesindex = genusCountsandMetadata$SampleID[genusCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground %in% micegroups_all]
micegroups_inflamed.samplesindex = genusCountsandMetadata$SampleID[genusCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground %in% micegroups_inflamed]
micegroups_noninflamed.samplesindex = genusCountsandMetadata$SampleID[genusCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground %in% micegroups_noninflamed]
excludcrossoverGroup.samplesindex = genusCountsandMetadata$SampleID[genusCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground %in% excludecrossoverGroup]
micegroups_inflamed_withbothHMCohorts.samplesindex = genusCountsandMetadata$SampleID[genusCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground %in% micegroups_inflamed_withbothHMCohorts]
micegroups_serialTransfersPlusCrossover.samplesindex = genusCountsandMetadata$SampleID[genusCountsandMetadata$FMTGroupFMTsourcegtRecipientbackground %in% micegroups_serialTransfersPlusCrossover]

micegroups_all.metagroup = factor(metagroup[metagroup %in% micegroups_all])
micegroups_inflamed.metagroup = factor(metagroup[metagroup %in% micegroups_inflamed])
micegroups_noninflamed.metagroup = factor(metagroup[metagroup %in% micegroups_noninflamed])
excludecrossoverGroup.metagroup = factor(metagroup[metagroup %in% excludecrossoverGroup])
micegroups_inflamed_withbothHMCohorts.metagroup = factor(metagroup[metagroup %in% micegroups_inflamed_withbothHMCohorts])
micegroups_serialTransfersPlusCrossover.metagroup = factor(metagroup[metagroup %in% micegroups_serialTransfersPlusCrossover])
## info set to generate pcoa plot 
  ## all mice groups
micegroups_all_genusCounts_samplesOnly_lognorm_df = genusCounts_samplesOnly_df_lognorm[rownames(genusCounts_samplesOnly_df_lognorm) %in% micegroups_all.samplesindex,]
micegroups_all.metagroup
micegroups_all_colors = plotColors_ver2[levels(metagroup) %in% micegroups_all]
  ## KO mice groups
micegroups_inflamed_genusCounts_samplesOnly_lognorm_df = genusCounts_samplesOnly_df_lognorm[rownames(genusCounts_samplesOnly_df_lognorm) %in% micegroups_inflamed.samplesindex,]
micegroups_inflamed.metagroup
micegroups_inflamed_colors = plotColors_ver2[levels(metagroup) %in% micegroups_inflamed]
  ## WT mice groups
micegroups_noninflamed_genusCounts_samplesOnly_lognorm_df = genusCounts_samplesOnly_df_lognorm[rownames(genusCounts_samplesOnly_df_lognorm) %in% micegroups_noninflamed.samplesindex,]
micegroups_noninflamed.metagroup
micegroups_noninflamed_colors = plotColors_ver2[levels(metagroup) %in% micegroups_noninflamed]
  ## excluding crossover group
excludecrossoverGroup_genusCounts_samplesOnly_lognorm_df = genusCounts_samplesOnly_df_lognorm[rownames(genusCounts_samplesOnly_df_lognorm) %in% excludcrossoverGroup.samplesindex,]
excludecrossoverGroup.metagroup
excludecrossoverGroup_colors = plotColors_ver2[levels(metagroup) %in% excludecrossoverGroup]
  ## KO mice groups with both HM cohorts
micegroups_inflamed_withbothHMCohorts_genusCounts_samplesOnly_lognorm_df = genusCounts_samplesOnly_df_lognorm[rownames(genusCounts_samplesOnly_df_lognorm) %in% micegroups_inflamed_withbothHMCohorts.samplesindex,]
micegroups_inflamed_withbothHMCohorts.metagroup
micegroups_inflamed_withbothHMCohorts_colors = plotColors_ver2[levels(metagroup) %in% micegroups_inflamed_withbothHMCohorts]

## serial transfer groups plus crossover group
micegroups_serialTransfersPlusCrossover_genusCounts_samplesOnly_lognorm_df = genusCounts_samplesOnly_df_lognorm[rownames(genusCounts_samplesOnly_df_lognorm) %in% micegroups_serialTransfersPlusCrossover.samplesindex,]
micegroups_serialTransfersPlusCrossover.metagroup
micegroups_serialTransfersPlusCrossover_colors = plotColors_ver2[levels(metagroup) %in% micegroups_serialTransfersPlusCrossover]

pdf("./figures/03_06_2023_edits/micegroups_serialTransfersPlusCrossover.pdf")
generatePCOAPlots(micegroups_serialTransfersPlusCrossover_genusCounts_samplesOnly_lognorm_df, 
                  micegroups_serialTransfersPlusCrossover.metagroup,
                  micegroups_serialTransfersPlusCrossover_colors, legendlocation = "topright", 
                  plotTitle="Mice Groups, Including Crossover Group")

dev.off()


generatePCOAPlots(micegroups_inflamed_withbothHMCohorts_genusCounts_samplesOnly_lognorm_df, 
                  micegroups_inflamed_withbothHMCohorts.metagroup,
                  micegroups_inflamed_withbothHMCohorts_colors, legendlocation = "topleft", plotTitle="Inflamed Mice Groups, Both HM Cohorts")

generatePCOAPlots(genusCounts_samplesOnly_df_lognorm, metagroup,
                  plotColors_ver2, legendlocation = "bottomleft", plotTitle="All Groups")

generatePCOAPlots(micegroups_all_genusCounts_samplesOnly_lognorm_df, micegroups_all.metagroup,
                  micegroups_all_colors, legendlocation = "topright", plotTitle="Mice Groups Only")

generatePCOAPlots(excludecrossoverGroup_genusCounts_samplesOnly_lognorm_df, excludecrossoverGroup.metagroup,
                  excludecrossoverGroup_colors, legendlocation = "topleft", plotTitle="Excluding Cross-Over Group**")

generatePCOAPlots(micegroups_noninflamed_genusCounts_samplesOnly_lognorm_df, micegroups_noninflamed.metagroup,
                  micegroups_noninflamed_colors, legendlocation = "topright", plotTitle="Non-inflamed Mice Groups")

generatePCOAPlots(micegroups_inflamed_genusCounts_samplesOnly_lognorm_df, micegroups_inflamed.metagroup,
                  micegroups_inflamed_colors, legendlocation = "topright", plotTitle="Inflamed Mice Groups")



```

### Correlations and Scatterplots

The code below generates the following:

-   Correlation coefficients for sample-pairs 1) within group and 2)across groups
-   Boxplots that summarize the correlation coefficients from the prior (above) step
-   Scatterplots for each sample-pair comparison, which reflects ASV abundance between a given pair of samples

The code below generates the ASV scatterplots and boxplots for samples WITHIN groups with the updated nomenclature and specific interests in the serial passage groups.

```{r}
## FILES
file_asv_updatedName_t_samplesOnly_lognorm #samples only, lognorm, samples x feature
labels_inputRecipientGroups #all labels
# serial passage groups
labels_NIMM_lineage = c("HM1input", "HM1->WT", "NIMM-g1input", "NIMM-g1->WT", "NIMM-g2input", "NIMM-g2->WT")
labels_IMM_lineage = c("HM1input", "HM1->KO", "IMM-g1input", "IMM-g1->KO","IMM-g2input", "IMM-g2->KO")
labels_serialpassage_newNames = c("HM1input", "HM1->WT", "NIMM-g1input", "NIMM-g1->WT", "NIMM-g2input", "NIMM-g2->WT","HM1->KO", "IMM-g1input", "IMM-g1->KO","IMM-g2input", "IMM-g2->KO")
myASVandMetadata_newNames#counts table with metadata, not lognorm, samples x feature
names(myASVandMetadata_newNames)[names(myASVandMetadata_newNames)=="FMTGroupFMTsource..Recipientbackground"] = "FMTGroupFMTsourcegtRecipientbackground"
## scatterplots

  #within groups
withinGroup_pearson_newNames_serialpassage=getCorrelationCoefficient_withinGroup(groupList=labels_serialpassage_newNames, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson")

withinGroup_pearson_newNames_NIMMlineage=getCorrelationCoefficient_withinGroup(groupList=labels_NIMM_lineage, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson")

withinGroup_pearson_newNames_IMMlineage=getCorrelationCoefficient_withinGroup(groupList=labels_IMM_lineage, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson")


## figure 4s
withinGroup_pearson_IMMlineage_genusCounts =getCorrelationCoefficient_withinGroup(groupList=labels_IMM_lineage, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson")


## create pdf of boxplots 
pdf("/Users/aduong/Desktop/Balfour2022/figureset_17Dec2022/asv_scatterplot_withinGroup_allSerialPassage.pdf", height = 8, width = 10)
boxplot(withinGroup_pearson_newNames_serialpassage,names = labels_serialpassage_newNames, las=2, cex.axis=0.70,
        main="Correlation Coefficients Within Groups, All Serial Passage Groups", cex.main=0.8)
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figureset_17Dec2022/asv_scatterplot_withinGroup_SerialPassage_NIMM.pdf", height = 8, width = 10)
boxplot(withinGroup_pearson_newNames_NIMMlineage,names = labels_NIMM_lineage, las=2, cex.axis=0.70,
        main="Correlation Coefficients Within Groups, Serial Passage for NIMM Lineage", cex.main=0.8)
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figureset_17Dec2022/asv_scatterplot_withinGroup_SerialPassage_IMM.pdf", height = 8, width = 10)
boxplot(withinGroup_pearson_newNames_IMMlineage,names = labels_IMM_lineage, las=2, cex.axis=0.70,
        main="Correlation Coefficients Within Groups, Serial Passage for IMM Lineage", cex.main=0.8)
dev.off()

## create scatter plots for asv abudnance, samples across groups
generateScatterPlots_samplePairsAcrossGroup(groupList=labels_serialpassage_newNames, tableWithMeta=myASVandMetadata_newNames,corrTestMethod = "pearson"  ,tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, resultsDirectory = "/Users/aduong/Desktop/Balfour2022/figures/figuresAndTables_jan2023/asvAbundance_scatterplots/acrossGroups_pearson/")

```

#### ASV Transfer Efficiencies Within FMT Groups

Next, we generate correlations (within groups) for samples within only human cohorts, which are then visually summarized as boxplots.

```{r}

 ## only human cohorts
labels_bothHumanCohorts = c("HM1->KO", "HM2->KO")

withinGroup_pearson_humanCohortsOnly = getCorrelationCoefficient_withinGroup(groupList=labels_bothHumanCohorts, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson", variableName = "FMTGroupFMTsourcegtRecipientbackground", sampleColumnName = "SampleID")

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/boxplots/ASV_boxplot_withinGroup_bothHumanCohortsOnly.pdf")
boxplot(withinGroup_pearson_humanCohortsOnly,names = labels_bothHumanCohorts, las=2, cex.axis=0.70,
        main="Pearson Correlation Coefficients Within Groups, Human Cohorts 1 and 2", cex.main=0.8)
dev.off()


```



#### ASV Transfer Efficiencies Across FMT Groups

The code below generates asv scatterplots and boxplots for samples ACROSS groups with the updated nomenclature and specific interests in the serial passage groups.

```{r}
## FILES
file_asv_updatedName_t_samplesOnly_lognorm #samples only, lognorm, samples x feature
labels_inputRecipientGroups #all labels
# serial passage groups
labels_NIMM_lineage = c("HM1input", "HM1->WT", "NIMM-g1input", "NIMM-g1->WT", "NIMM-g2input", "NIMM-g2->WT")
labels_IMM_lineage = c("HM1input", "HM1->KO", "IMM-g1input", "IMM-g1->KO","IMM-g2input", "IMM-g2->KO")
labels_serialpassage_newNames = c("HM1input", "HM1->WT", "NIMM-g1input", "NIMM-g1->WT", "NIMM-g2input", "NIMM-g2->WT","HM1->KO", "IMM-g1input", "IMM-g1->KO","IMM-g2input", "IMM-g2->KO")
myASVandMetadata_newNames#counts table with metadata, not lognorm, samples x feature

#extract groups of interest for asv correlation boxplots
fig4a.wtlineage.index = c(1,20,35)
fig4a.wtlineage.supp.index = c(2,4,21)

fig4b.kolineage.index = c(6,50,55)
fig4b.kolineage.supp.index = c(7,9,51)

fig4c.index = c(12,29,47,54)
fig4c.supp.index = c(32,43,34,45)

```

Generate the correlations between samples from across groups.

```{r}
#across groups
corr_acrossGroups_pearson_asvCounts_withHuman = getCorrelationCoefficient_acrossGroup(groupList=labels_serialpassage_newNames, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson")

corr_acrossGroups_pearson_GenusCounts_withHuman = getCorrelationCoefficient_acrossGroup(groupList=labels_serialpassage_newNames, tableWithMeta=genusCountsandMetadata, tableCountsOnly=genusCounts_samplesOnly_df_lognorm, 
corrTestMethod="pearson", variableName = "FMTGroupFMTsourcegtRecipientbackground", sampleColumnName = "SampleID")


    ## with phylum level counts
corr_acrossGroups_pearson_PhylumCounts_withHuman = getCorrelationCoefficient_acrossGroup(groupList=labels_serialpassage_newNames, tableWithMeta=phylumCountsandMetadata, tableCountsOnly=phylumCounts_samplesOnly_df_lognorm, 
corrTestMethod="pearson", variableName = "FMTGroupFMTsourcegtRecipientbackground", sampleColumnName = "SampleID")

boxplot(corr_acrossGroups_pearson_GenusCounts_withHuman[[2]])

all.equal(corr_acrossGroups_pearson_GenusCounts_withHuman[[1]], corr_acrossGroups_pearson_PhylumCounts_withHuman[[1]])

```

These boxplots are for inflamed (KO) groups in our SV transfer comparison across groups.

```{r}
  #asv level counts
pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_inflamed_asv.pdf")
boxplot(corr_acrossGroups_pearson_asvCounts_withHuman[[2]][fig4b.kolineage.index],names = corr_acrossGroups_pearson_asvCounts_withHuman[[1]][fig4b.kolineage.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Inflamed Groups", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_inflamed_asv_supp.pdf")
boxplot(corr_acrossGroups_pearson_asvCounts_withHuman[[2]][fig4b.kolineage.supp.index],names = corr_acrossGroups_pearson_asvCounts_withHuman[[1]][fig4b.kolineage.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Inflamed Groups (SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

  # genus level counts 
corr_acrossGroups_pearson_GenusCounts_withHuman
pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_inflamed_genus.pdf")
boxplot(corr_acrossGroups_pearson_GenusCounts_withHuman[[2]][fig4b.kolineage.index],names = corr_acrossGroups_pearson_GenusCounts_withHuman[[1]][fig4b.kolineage.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Inflamed Groups (Genus)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_inflamed_genus_supp.pdf")
boxplot(corr_acrossGroups_pearson_GenusCounts_withHuman[[2]][fig4b.kolineage.supp.index],names = corr_acrossGroups_pearson_GenusCounts_withHuman[[1]][fig4b.kolineage.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Inflamed Groups (Genus) (SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

    #phylum counts
corr_acrossGroups_pearson_PhylumCounts_withHuman

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_inflamed_phylum.pdf")
boxplot(corr_acrossGroups_pearson_PhylumCounts_withHuman[[2]][fig4b.kolineage.index],names = corr_acrossGroups_pearson_PhylumCounts_withHuman[[1]][fig4b.kolineage.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Inflamed Groups (Phylum)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_inflamed_phylum_supp.pdf")
boxplot(corr_acrossGroups_pearson_PhylumCounts_withHuman[[2]][fig4b.kolineage.supp.index],names = corr_acrossGroups_pearson_PhylumCounts_withHuman[[1]][fig4b.kolineage.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Inflamed Groups (Phylum, SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

```

These boxplots are for non-inflamed (WT) groups in our SV transfer comparison across groups.

```{r}
pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_NONinflamed_asv.pdf")
boxplot(corr_acrossGroups_pearson_asvCounts_withHuman[[2]][fig4a.wtlineage.index],names = corr_acrossGroups_pearson_asvCounts_withHuman[[1]][fig4a.wtlineage.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Non-inflamed Groups", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_NONinflamed_asv_supp.pdf")
boxplot(corr_acrossGroups_pearson_asvCounts_withHuman[[2]][fig4a.wtlineage.supp.index],names = corr_acrossGroups_pearson_asvCounts_withHuman[[1]][fig4a.wtlineage.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Non-inflamed Groups (SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

    #genus counts
corr_acrossGroups_pearson_GenusCounts_withHuman
pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_NONinflamed_genus.pdf")
boxplot(corr_acrossGroups_pearson_GenusCounts_withHuman[[2]][fig4a.wtlineage.index],names = corr_acrossGroups_pearson_GenusCounts_withHuman[[1]][fig4a.wtlineage.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Non-inflamed Groups (Genus)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_NONinflamed_genus_supp.pdf")
boxplot(corr_acrossGroups_pearson_GenusCounts_withHuman[[2]][fig4a.wtlineage.supp.index],names = corr_acrossGroups_pearson_GenusCounts_withHuman[[1]][fig4a.wtlineage.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Non-inflamed Groups (Genus,SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

  # phylum counts
corr_acrossGroups_pearson_PhylumCounts_withHuman
pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_NONinflamed_phylum.pdf")
boxplot(corr_acrossGroups_pearson_PhylumCounts_withHuman[[2]][fig4a.wtlineage.index],names = corr_acrossGroups_pearson_PhylumCounts_withHuman[[1]][fig4a.wtlineage.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Non-inflamed Groups (Phylum)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_NONinflamed_phylum_supp.pdf")
boxplot(corr_acrossGroups_pearson_PhylumCounts_withHuman[[2]][fig4a.wtlineage.supp.index],names = corr_acrossGroups_pearson_PhylumCounts_withHuman[[1]][fig4a.wtlineage.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Non-inflamed Groups (Phylum,SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

```

These boxplots include the cross-over groups in our SV transfer comparison across groups.

```{r}
pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_crossover_asv.pdf", width = 10)
boxplot(corr_acrossGroups_pearson_asvCounts_withHuman[[2]][fig4c.index],names = corr_acrossGroups_pearson_asvCounts_withHuman[[1]][fig4c.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Cross-over Group", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_crossover_supp_asv.pdf")
boxplot(corr_acrossGroups_pearson_asvCounts_withHuman[[2]][fig4c.supp.index],names = corr_acrossGroups_pearson_asvCounts_withHuman[[1]][fig4c.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Cross-over Group (SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

    #genus counts
corr_acrossGroups_pearson_GenusCounts_withHuman
pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_crossover_genus.pdf")
boxplot(corr_acrossGroups_pearson_GenusCounts_withHuman[[2]][fig4c.index],
        names = corr_acrossGroups_pearson_GenusCounts_withHuman[[1]][fig4c.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Cross-over Group (Genus)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_crossover_supp_genus.pdf")
boxplot(corr_acrossGroups_pearson_GenusCounts_withHuman[[2]][fig4c.supp.index],names = corr_acrossGroups_pearson_GenusCounts_withHuman[[1]][fig4c.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Cross-over Group (Genus, SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

    #phylum counts 
corr_acrossGroups_pearson_PhylumCounts_withHuman

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_crossover_phylum.pdf")
boxplot(corr_acrossGroups_pearson_PhylumCounts_withHuman[[2]][fig4c.index],
        names = corr_acrossGroups_pearson_PhylumCounts_withHuman[[1]][fig4c.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Cross-over Group (Phylum)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

pdf("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/fig4_crossover_supp_phylum.pdf")
boxplot(corr_acrossGroups_pearson_PhylumCounts_withHuman[[2]][fig4c.supp.index],names = corr_acrossGroups_pearson_PhylumCounts_withHuman[[1]][fig4c.supp.index],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, Cross-over Group (Phylum, SUPPFIG)", cex.main=0.8,
        ylim=c(0,1))
dev.off()

```

These boxplots are for all serial passage groups in our SV transfer comparison across groups. We did not include this visual in the manuscript; this is just for an FYI.

```{r}
boxplot(corr_acrossGroups_pearson_asvCounts_withHuman[[2]],names = corr_acrossGroups_pearson_asvCounts_withHuman[[1]],  cex.axis=0.6,
        main="Correlation Coefficients Across Groups, All Serial Passage Groups", cex.main=0.8,
        ylim=c(0,1))
```

#### ASV Transfer Efficiencies Across FMT Groups: Non-transferred ASVs

Here, we plot the abundance of the non-transferred ASVs.

```{r}

#metadata for asv and taxonomy
asv_taxonomy_map

my_ntasv_vectors = samplePairsAcrossGroup_NontransferringASVS_returnVectors(groupList=labels_serialpassage_newNames, 
                                                                         tableWithMeta=myASVandMetadata_newNames, 
                                                                         tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, 
                                                                         sampleColumnName = "SampleID",
                                                                         variableName = "FMTGroupFMTsourcegtRecipientbackground")

my_ntASV_labels = samplePairsAcrossGroup_NontransferringASVS_returnLabels(groupList=labels_serialpassage_newNames,
                                             tableWithMeta=myASVandMetadata_newNames,
                                             tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm) 

for(i in 1:length(my_ntasv_vectors)){
  h1 = hist(unlist(my_ntasv_vectors[[i]][1]),
            breaks=seq(0,5,length.out=10))
  h2 = hist(unlist(my_ntasv_vectors[[i]][2]),
            breaks=seq(0,5,length.out=10))
  pdf(paste0("/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/histograms_nontransferredASV_march2023/set5_04_21_23/",
             my_ntASV_labels[[i]],".pdf"))
  plot(h1, col="red",xlim=c(0,5),ylim=c(0,120),)
  plot(h2, add = T, col=scales::alpha("blue",.6),xlim=c(0,5), ylim=c(0,120))
  dev.off()
  
}

generateTableWithTaxonomy_NontransferringASVS_samplePairsAcrossGroup(groupList=labels_serialpassage_newNames,
                                                                     taxonomyMap=asv_taxonomy_map,
                                                                     taxaColName="taxonomy",
                                                                     asvColName="OTUID",
                                                                     tableWithMeta=myASVandMetadata_newNames,
                                                                     tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm,
                                                                     resultsDirectory="/Users/aduong/Desktop/Balfour2022/figures/03_06_2023_edits/histograms_nontransferredASV_march2023/tablesWithTaxonomy_04_21_2023/")


```

#### ASV Transfer Efficiencies Within FMT Groups at the Phylum level

The code below parses the taxonomy column to count the number of different phlum present in this counts table. The phylum "NA" means unclassified.

There are a total of 13 phylum in the ASV counts table. Of these 13, four only had one ASV present. For statistical testing purposes, these phylum were excluded from the Pearson Correlation Coefficient test: Arthropoda, Cyanobacteria, Desulfobacterota, and Vertebrata. Campilobacterota and Patescibacteria were very sparse, present in 5 or less samples each; this led to empty results in the boxplot, so we excluded these two as well.

```{r}

## ver 1 parsing phylum name
taxanames_all = file_asv_updatedName$taxonomy

taxanames_all_split = strsplit(taxanames_all, ";")

phylum_label = NULL

for(i in 1:length(taxanames_all_split)){
  phylum_label[i] = taxanames_all_split[[i]][2]
}

phylum_label_short = gsub("p__", "", phylum_label)
phylum_label_short = gsub(" ", "", phylum_label_short)

mapping_ASVandPhyla = data.frame("ASV" = file_asv_updatedName$OTUID, "PhylumLabels" = phylum_label_short)
table(mapping_ASVandPhyla$PhylumLabels)

mapping_ASVandPhyla$PhylumLabels[is.na(mapping_ASVandPhyla$PhylumLabels)] = "Unclassified"

#exclude the single ASV phylum
phlyum_groups = unique(mapping_ASVandPhyla$PhylumLabels)
singles = c("Arthropoda","Cyanobacteria","Desulfobacterota", "Vertebrata")
phlyum_groups_subset = phlyum_groups[! phlyum_groups %in% singles]

#rename the NA group in phylum list 
phlyum_groups_subset_2 = phlyum_groups_subset
phlyum_groups_subset_2[7] = "Unclassified"

phlyum_groups_subset_withoutSparsePhyla = phlyum_groups_subset_2[phlyum_groups_subset_2 != "Patescibacteria" & phlyum_groups_subset_2 != "Campilobacterota"]

phlyum_groups_subset_withoutSparsePhyla[7] = "Unclassified"

mapping_ASVandPhyla


myGroup_serialPassage_miceOnly = c("IMM-g1->KO", "IMM-g2->KO", "NIMM-g1->WT", "NIMM-g2->WT")
myGroup_serialPassage_miceandHuman = c("IMM-g1->KO", "IMM-g2->KO", "NIMM-g1->WT", "NIMM-g2->WT", "HM1->KO", "HM1->WT")
sampleID_for_myGroup_serialPassage_miceOnly = metamapping_updatedName$SampleID[metamapping_updatedName$FMTGroupFMTsource..Recipientbackground %in% myGroup_serialPassage_miceOnly[1]]

mygroup_outfile_labels = gsub("->", "_to_", myGroup_serialPassage_miceandHuman)

## for all in the serial passage list
for(i in 1:length(myGroup_serialPassage_miceandHuman)){
  
  myCorr_withinGroup_pearson_phlyum_perIRGroup_OUTPUT=getCorrelationCoefficient_withinGroup_phylumLevel_byGroup(groupList=phlyum_groups_subset_withoutSparsePhyla, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson", sampleGroup = myGroup_serialPassage_miceandHuman[i], mappingMetaFile = mapping_ASVandPhyla,mappingMetaFile_taxaColName = "ASV", mappingMetaFile_phylumColName = "PhylumLabels", tableWithMeta_sampleColName = "SampleID" ,tableWithMeta_groupColName = "FMTGroupFMTsourcegtRecipientbackground")
  
  pdf(paste0("/Users/aduong/Desktop/Balfour2022/figures/april_checkingTests/withinGroup_PearsonCorr_phylumLevel_",mygroup_outfile_labels[i] , ".pdf"), width = 10)
  boxplot(myCorr_withinGroup_pearson_phlyum_perIRGroup_OUTPUT[[2]], names = phlyum_groups_subset_withoutSparsePhyla, main=paste0("Pearson Correlation Coefficient (R), Phylum Level ", myGroup_serialPassage_miceandHuman[i]), cex.axis=0.65, yaxt="n")
  axis(side=2, at=seq(0, 1, by=0.1))
  dev.off()
  
}


```

Next, we calculate t-test p-values to compare the correlations for each phylum between WT and KO groups to determine if there are significant differences between phylum for each phenotype.

At this time, I performed the t-test on the HM1-\>WT vs HM1-\>KO group. Results are generated as a dataframe with unadjusted p-values for each phylum. This dataframe is written out as a tab-delimited text file.

```{r}
## ttest to compare wt and ko phylum groups

myCorr_withinGroup_pearson_phlyum_immg1toKO=getCorrelationCoefficient_withinGroup_phylumLevel_byGroup(groupList=phlyum_groups_subset_withoutSparsePhyla, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson", sampleGroup = myGroup_serialPassage_miceandHuman[1], mappingMetaFile = mapping_ASVandPhyla,mappingMetaFile_taxaColName = "ASV", mappingMetaFile_phylumColName = "PhylumLabels", tableWithMeta_sampleColName = "SampleID" ,tableWithMeta_groupColName = "FMTGroupFMTsourcegtRecipientbackground")

myCorr_withinGroup_pearson_phlyum_nimmg1toWT=getCorrelationCoefficient_withinGroup_phylumLevel_byGroup(groupList=phlyum_groups_subset_withoutSparsePhyla, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson", sampleGroup = myGroup_serialPassage_miceandHuman[3], mappingMetaFile = mapping_ASVandPhyla, mappingMetaFile_taxaColName = "ASV", mappingMetaFile_phylumColName = "PhylumLabels", tableWithMeta_sampleColName = "SampleID" ,tableWithMeta_groupColName = "FMTGroupFMTsourcegtRecipientbackground")

myCorr_withinGroup_pearson_phlyum_hm1toWT=getCorrelationCoefficient_withinGroup_phylumLevel_byGroup(groupList=phlyum_groups_subset_withoutSparsePhyla, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson", sampleGroup = myGroup_serialPassage_miceandHuman[6], mappingMetaFile = mapping_ASVandPhyla,mappingMetaFile_taxaColName = "ASV", mappingMetaFile_phylumColName = "PhylumLabels", tableWithMeta_sampleColName = "SampleID" ,tableWithMeta_groupColName = "FMTGroupFMTsourcegtRecipientbackground")

myCorr_withinGroup_pearson_phlyum_hm1toKO=getCorrelationCoefficient_withinGroup_phylumLevel_byGroup(groupList=phlyum_groups_subset_withoutSparsePhyla, tableWithMeta=myASVandMetadata_newNames, tableCountsOnly=file_asv_updatedName_t_samplesOnly_lognorm, corrTestMethod="pearson", sampleGroup = myGroup_serialPassage_miceandHuman[5], mappingMetaFile = mapping_ASVandPhyla,mappingMetaFile_taxaColName = "ASV", mappingMetaFile_phylumColName = "PhylumLabels", tableWithMeta_sampleColName = "SampleID" ,tableWithMeta_groupColName = "FMTGroupFMTsourcegtRecipientbackground")


ttestpval_list = NULL
for(i in 1:length(phlyum_groups_subset_withoutSparsePhyla)){

  a = unlist(myCorr_withinGroup_pearson_phlyum_hm1toWT[[2]][i])
  b = unlist(myCorr_withinGroup_pearson_phlyum_hm1toKO[[2]][i]) 
  if(sum(is.na(a)) >=1 | sum(is.na(b)) >=1){print(paste0(phlyum_groups_subset_withoutSparsePhyla[i], " : NA"))} else {

  ttest_pval = t.test(a,b)$p.value
  resultstoprint = paste0(phlyum_groups_subset_withoutSparsePhyla[i],
                          ": ", ttest_pval)
  ttestpval_list[i] = ttest_pval
  
  }
}
   
mydf = data.frame("Phylum"=phlyum_groups_subset_withoutSparsePhyla, 
                    "Pvalue" = ttestpval_list)

write.table(mydf, "/Users/aduong/Desktop/Balfour2022/figures/april_checkingTests/phylumboxplot_ttest_hm1toWT_vs_hm1toKO.txt", sep = "\t", quote = F, row.names = F)

```

#### Testing homogeneity of variances

Bartlett's test checks for equality of variances between two or more groups.

Levene's test is the non-parametric version of variance equality testing.

After testing the data for normality (Shapiro-Wilkes test), the data presented with a normal distribution.

However, there were some outlier scores within some groups, so we also performed the non-parametric Levene's test for a more robust asseessment of the histology scores.

```{r}
library(stats) # for bartletts test
library(car) #for levene test


histologyScores_hm1immg1immg2toKO = read.table("/Users/aduong/Desktop/Balfour2022/files/histologyScores_hm1immg1immg2toKO.txt", sep = "\t", header = T)

phenotype_groups = unique(histologyScores_hm1immg1immg2toKO$Group)
colnames(histologyScores_hm1immg1immg2toKO)
regions_list = c("Ileum", "Cecum","Proximal.Colon","Distal.Colon","Rectum")

bartlettResults_list = list()
for(i in 1:length(regions_list)){
  bartlettPvalues= calculateBartlettPvalue_histologyScoreData(histologyScores_hm1immg1immg2toKO,
                                                           regionColName = regions_list[i],      groupColName = "Group", phenotypeNames=phenotype_groups)
  
  bartlettResults_list[[i]]= bartlettPvalues
}

names(bartlettResults_list)=regions_list

  ## running Barlett's test on the total histology score

barlettResult_totalColonScore = calculateBartlettPvalue_histologyScoreData(histologyScores_hm1immg1immg2toKO,
                                             regionColName = "Total.1", 
                                             groupColName = "Group", phenotypeNames=phenotype_groups)

levenesTest_totalColonScore=calculateLevenesTestPvalue_histologyScoreData(histologyScores_hm1immg1immg2toKO,regionColName = "Total.1", groupColName = "Group",phenotypeNames=phenotype_groups)


colnames(histologyScores_hm1immg1immg2toKO)

## run Levene's Test
leveneResults_list = list()
for(i in 1:length(regions_list)){
  levenePvalues= calculateLevenesTestPvalue_histologyScoreData(histologyScores_hm1immg1immg2toKO,
                                                           regionColName = regions_list[i], groupColName = "Group",phenotypeNames=phenotype_groups)
  
  leveneResults_list[[i]]= levenePvalues
}

names(leveneResults_list)=regions_list




## run KS test
ksResults_perGroup = list()

for(i in 1:length(regions_list)){
hm1_to_g1 = ks.test(subset_df_list[[1]][,regions_list[i]],
        subset_df_list[[2]][,regions_list[i]])

hm1_to_g2 =ks.test(subset_df_list[[1]][,regions_list[i]],
        subset_df_list[[3]][,regions_list[i]])

g1_vs_g2 = ks.test(subset_df_list[[2]][,regions_list[i]],
        subset_df_list[[3]][,regions_list[i]])

ksResults_perGroup[[i]] = list("hm1_to_g1"=hm1_to_g1,
                               "hm1_to_g2"=hm1_to_g2,
                               "g1_vs_g2"=g1_vs_g2)

}
names(ksResults_perGroup)=regions_list

```

#### Visualizing histology score data as boxplots

In addition to the provided barplots of the histology scores, we also generated this data as boxplots. This way, we can more easily view the spread of the datapoints for each region across the different FMT groups.

```{r}
## boxplot the histology score data

histologyScores_hm1immg1immg2toKO

histology_fmtList = unique(histologyScores_hm1immg1immg2toKO$Group)

subset_df_list = list()

for(i in 1:length(histology_fmtList)){

  df_byGroup = subset(histologyScores_hm1immg1immg2toKO, 
                      histologyScores_hm1immg1immg2toKO$Group==histology_fmtList[i])
  
  subset_df_list[[i]]=df_byGroup

}
names(subset_df_list)=histology_fmtList
regions_list

pdf("/Users/aduong/Desktop/Balfour2022/figures/histologyScores_boxplots.pdf",width = 12)
par(mfrow=c(2,5))
for(i in 1:length(regions_list)){
  boxplot(list(subset_df_list[[1]][,regions_list[i]], 
               subset_df_list[[2]][,regions_list[i]],
               subset_df_list[[3]][,regions_list[i]]),
          ylim=c(0,4), names=names(subset_df_list),
          main=regions_list[i], las=2)
} 
dev.off()

```

#### Cage Effect

We assess transfer inefficiencies for samples within the same groups per cage.

```{r}
    ## with the 6 fmt groups
metadataNames= c("SampleID","UniversalCageNumber","Background", "FMTGroupFMTsourcegtRecipientbackground", 
               "Passge", "Sex", "Agewk", "SampleType")
withinGroups_perCage = getCorrelationCoefficient_withinGroup_perCage(tableWithMeta=myASVandMetadata_newNames, 
                                                                 corrTestMethod="pearson", 
                                                                 phenotypeGroup=fmtList_withinGroup, 
                                                                 tableWithMeta_sampleColName="SampleID",
                                                                 tableWithMeta_groupColName="FMTGroupFMTsourcegtRecipientbackground",
                                                                 tableWithMeta_cageColName="UniversalCageNumber",
                                                                metadataNames = metadataNames)


par(mar=c(6, 4, 2, 2))
boxplot(withinGroups_perCage[[2]], names = withinGroups_perCage[[1]], las=2)
stripchart(withinGroups_perCage[[2]], method="jitter",add=T, pch=16, vertical = T)


```

#### 2D hierarchical clustering with relative taxonomic abundance

```{r}
library(factoextra)
library(cluster)


```

#### Testing

The code below creates the same figures as the ones generated through the Plotmicrobiome application.

The alpha diversity plots were generated through the Plotmicrobiome application (developed by Shan Sun); we tested these figures by independently recreating (with our own R scripts) these figures in R.

The PCoA results were initially generated through the Plotmicrobiome Application as well; we also independently recreated these plots for formatting purposes (text/plot size) and to check if we were able to recreate the same results as the results from the Plotmicrobiome app.

For transparency, we multiplied either one or both of the MDS axes in order to perfectly replicate the Plotmicrobiome plot results. PERMANOVA R and p-values, as well as MDS percent variance were the same as the Plotmicrobiome plot results.

```{r}
# par(mar=c(5, 4, 4, 2) + 0.1)


pdf("./figures/03_06_2023_edits/pcoaPlots_toMatchAppFigures.pdf")
  # all groups
par(mar=c(5, 5, 4, 2))
generatePCOAPlots(genusCounts_samplesOnly_df_lognorm, metagroup,
                  plotColors_ver2, legendlocation = "bottomleft", plotTitle="All Groups")

    # exclude crossover group
generatePCOAPlots_flipYaxis(excludecrossoverGroup_genusCounts_samplesOnly_lognorm_df, excludecrossoverGroup.metagroup,
                            excludecrossoverGroup_colors, legendlocation = "topleft", plotTitle="Excluding Cross-Over Group")
    # mice groups only
generatePCOAPlots_flipYaxis(micegroups_all_genusCounts_samplesOnly_lognorm_df, micegroups_all.metagroup,
                  micegroups_all_colors, legendlocation = "topright", plotTitle="Mice Groups Only")
  # inflamed mice only
generatePCOAPlots(micegroups_inflamed_genusCounts_samplesOnly_lognorm_df, micegroups_inflamed.metagroup,
                  micegroups_inflamed_colors, legendlocation = "topright", plotTitle="Inflamed Mice Groups")

  # non-inflamed mice only
generatePCOAPlots_flipBothaxes(micegroups_noninflamed_genusCounts_samplesOnly_lognorm_df, micegroups_noninflamed.metagroup,
                  micegroups_noninflamed_colors, legendlocation = "topleft", plotTitle="Non-inflamed Mice Groups")

dev.off()
```
